server/src/main/java/org/opensearch/search/internal/SearchContext.java

```diff
@@ -393,12 +393,16 @@ public final boolean hasOnlySuggest() {
     public abstract long getRelativeTimeInMillis();
```

Comment:
Should we deprecate this one and point to `getRelativeTimeInMillis(boolean useCache)` ? We definitely don't need 2 methods for same thing.

Reply:
@reta  if want to deprecate `getRelativeTimeInMillis` which is an abstract method then `getRelativeTimeInMillis(boolean useCache)` should be abstract too to give child classes ability to implement it and I think that not all child classes really should care about `useCache` because they might not have such concept so this will cause even more abstraction leak. while by keeping both overrides then child class has to implement one method/logic.  so I'm not sure what is best here as both options is causing abstraction leak anyway. do you mean by deprecation just annotating the old method with `@Deprecated` and leave it ?

Reply:
>  do you mean by deprecation just annotating the old method with @Deprecated and leave it ?

@aabukhalil yeah, exactly, mark as `@deprecated`, we cannot remove it right now

Reply:
@reta marking `getRelativeTimeInMillis()` as deprecated while still abstract and forcing child classes to implement it because there is no path forward to implement, does not make sense to me. Deprecation is not inherited in java so if you deprecate `getRelativeTimeInMillis()`  in `SearchContext` and you have subclass `MySearchContext` which extends `SearchContext` and you call  `getRelativeTimeInMillis()` on `MySearchContext` it won't be marked as deprecated which somehow breaks  Liskov substitution principle because the behavior of compiler will change if you have your variable defined as  `SearchContext` vs `MySearchContext`, what do you think ? I don't see strong reason to deprecate this.

Reply:
Oh this is true, `@Deprecated` indeed has an issue with inheritance, you are right @aabukhalil

Reply:
thank you @reta, appreciate your help and dedication :)